src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,43 +2,419 @@
+            return result;
+        }
+
+        public static unsafe Vector128<byte> CreateScalar(byte value)
+        {
+            var result = Vector128<byte>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<byte>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<double> CreateScalar(double value)
+        {
+            var result = Vector128<double>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<double>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<short> CreateScalar(short value)
+        {
+            var result = Vector128<short>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<short>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<int> CreateScalar(int value)
+        {
+            var result = Vector128<int>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<int>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<long> CreateScalar(long value)
+        {
+            var result = Vector128<long>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<long>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector128<sbyte> CreateScalar(sbyte value)
+        {
+            var result = Vector128<sbyte>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<sbyte>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<float> CreateScalar(float value)
+        {
+            var result = Vector128<float>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<float>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector128<ushort> CreateScalar(ushort value)
+        {
+            var result = Vector128<ushort>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<ushort>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector128<uint> CreateScalar(uint value)
+        {
+            var result = Vector128<uint>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<uint>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector128<ulong> CreateScalar(ulong value)
+        {
+            var result = Vector128<ulong>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<ulong>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<T> CreateScalar<T>(Vector64<T> value) where T : struct
```

Comment:
I opted for `CreateScalar` here (rather than something like `AsVector64`). It makes the API slightly more difficult to use, but I think more explicitly details that this is a non-zero cost operation (since it zero's the upper bits) and better allows us to expose the `Unsafe` version which leaves the upper bits uninitialized (and expose it in a consistent manner).